builtin/nomad/jobspec/platform.go

```diff
@@ -46,86 +54,314 @@ func (p *Platform) GenerationFunc() interface{} {
+	st.Update(fmt.Sprintf("Registering job %q...", *job.Name))
 	regResult, _, err := jobclient.Register(job, nil)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	s.Done()
+
+	// Store our state so we can destroy it properly
+	state.Name = result.Name
+	st.Step(terminal.StatusOK, "Job registration successful")
 
 	// Wait on the allocation
-	st := ui.Status()
-	defer st.Close()
 	evalID := regResult.EvalID
 	st.Update(fmt.Sprintf("Monitoring evaluation %q", evalID))
-	if err := nomad.NewMonitor(st, client).Monitor(evalID); err != nil {
-		return nil, err
+	if err := nomad.NewMonitor(st, client.NomadClient).Monitor(evalID); err != nil {
+		return err
 	}
 	st.Step(terminal.StatusOK, "Deployment successfully rolled out!")
 
-	return &result, nil
+	return nil
 }
 
-// Destroy deletes the Nomad job.
-func (p *Platform) Destroy(
+func (p *Platform) resourceJobDestroy(
+	ctx context.Context,
+	state *Resource_Job,
+	sg terminal.StepGroup,
+	client *nomadClient,
+) error {
+	step := sg.Add("")
+	defer func() { step.Abort() }()
+	step.Update("Deleting job: %s", state.Name)
+	step.Done()
+	_, _, err := client.NomadClient.Jobs().Deregister(state.Name, true, nil)
+	return err
+}
+
+func (p *Platform) resourceJobStatus(
 	ctx context.Context,
 	log hclog.Logger,
-	deployment *nomad.Deployment,
+	sg terminal.StepGroup,
+	state *Resource_Job,
+	client *nomadClient,
+	sr *resource.StatusResponse,
 	ui terminal.UI,
 ) error {
-	// We'll update the user in real time
+	s := sg.Add("Gathering health report for Nomad job...")
+	defer s.Abort()
+
+	jobClient := client.NomadClient.Jobs()
+
+	s.Update("Parsing the job specification...")
+	jobspec, err := p.jobspec(client.NomadClient, p.config.Jobspec)
+	if err != nil {
+		return err
+	}
+
+	jobResource := sdk.StatusReport_Resource{
+		Type:                "job",
+		CategoryDisplayHint: sdk.ResourceCategoryDisplayHint_INSTANCE_MANAGER,
+	}
+	sr.Resources = append(sr.Resources, &jobResource)
+
+	s.Update("Getting job info...")
+	q := &api.QueryOptions{Namespace: *jobspec.Namespace}
+	job, _, err := jobClient.Info(state.Name, q)
+	if err != nil {
+		return err
+	}
+
+	jobResource.Id = *job.ID
+	jobResource.Name = *job.Name
+	jobResource.CreatedTime = timestamppb.New(time.Unix(0, *job.SubmitTime))
+	stateJson, err := json.Marshal(map[string]interface{}{
+		"deployment": job,
+	})
+	jobResource.StateJson = string(stateJson)
+
+	// If job is running, start checking evals, then allocs
+	if *job.Status == structs.JobStatusRunning {
+		// Get list of evaluations for job
+		evals, _, err := jobClient.Evaluations(*job.ID, q)
+		evalStatus := "unblocked"
+		for _, eval := range evals {
+			switch eval.Status {
+			case structs.EvalStatusBlocked:
+				evalStatus = "blocked"
+				break
+			case structs.EvalStatusPending:
+				break
+			case structs.EvalStatusComplete:
+				break
+			case structs.EvalStatusFailed:
+				break
+			case structs.EvalStatusCancelled:
+				break
+			default:
+				evalStatus = "unknown"
+			}
+		}
```

Comment:
Could this be squashed down to a `hasSquashedEvals` bool? That'd make it more clear what the purpose of this block is and let the check below just be on that bool.

Reply:
@tgross will do! I originally wrote this with a bool, but I changed to using a string to account for the unknown case, so we could report an unknown status if there were a new Nomad eval status introduced but not handled here.